### ISSUE TITLE
segfault libobjc4

### ISSUE THREAD

**Issue by davedastrange** [threerings/openvpn-auth-ldap#60]
#uname -a
Linux 09e25e1e8cd7 4.4.0-45-generic #66-Ubuntu SMP Wed Oct 19 14:12:37 UTC 2016 x86_64 Linux

when starting the openvpn service it results in a sewgfault
segfault at 0 ip 00007fa790601af1 sp 00007fffc6cf8520 error 4 in libobjc.so.4.0.0[7fa7905f7000+20f000]

**Comment by claudio-walser** [comment]
I am having the exact same problem.
My openvpn runs in a docker container with alpine linux and starts well without auth-ldap activated. I search for a few days now and not making any pogress :(
Here is some strace output: https://gist.github.com/claudio-walser/fcba7842beb04070633ec1bf6f76980a
In dmesg i also have the same error with libobjc: [856394.633688] openvpn[11465]: segfault at 0 ip 00007faea4a2676f sp 00007ffd511973e0 error 4 in libobjc.so.4.0.0[7faea4a1c000+20f000]

**Comment by benley** [comment]
Related debian bug from 2011: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=641811

Their fix at the time was to build openvpn-auth-ldap with gcc 4.6 instead of 4.7.

**Comment by k0ste** [comment]
Try to apply [auth-ldap-2.0.3-gnustep.patch](https://aur.archlinux.org/cgit/aur.git/tree/auth-ldap-2.0.3-gnustep.patch?h=openvpn-auth-ldap). And build [--with-objc-runtime=modern](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=openvpn-auth-ldap#n45).

**Comment by nemaniarjun** [comment]
Did this get fixed?

**Comment by ivanovUA** [comment]
@nemaniarjun no, still the same. But I found way how to make it works. You can find it in my repo [https://github.com/ivanovUA/docker-openVPN/blob/master/Dockerfile](url)
